### PR TITLE
Check cairo contracts formatting

### LINKED ISSUE #1457: Check contracts formatting
### Feature Request

Check cairo contracts formatting in workflow by using `scarb fmt --check`.

### PR DESCRIPTION
**Pull request by franciszekjob**
Closes #1457 

## Introduced changes

Check cairo contracts formatting using `scarb fmt --check`

##

- [ ] This PR contains breaking changes

### CODE REVIEW

File: starknet_py/tests/e2e/mock/compile_contracts.sh
Comment (franciszekjob) [inline starknet_py/tests/e2e/mock/compile_contracts.sh:1]: note: I didn't add formatting check for cairo 0 contracts as they will be completely removed when https://github.com/software-mansion/starknet.py/issues/1334 is completed.

**Review by kkawula — COMMENTED** [review]
Would it be possible to add some kind of message why compiling failed? I suppose that it can be difficult to recognize the reason for test failure without a message. Especially when you are new to a project or external contributor

**Comment by franciszekjob** [comment]
> Would it be possible to add some kind of message why compiling failed? I suppose that it can be difficult to recognize the reason for test failure without a message. Especially when you are new to a project or external contributor

I had it on my mind, but I decided to leave it with default formatting error that `scarb` displays (same when compilation fails due to any error, we don't handle it in script). Wdyt @kkawula ? Don't have a strong opinion though.

**Comment by franciszekjob** [comment]
> > > Would it be possible to add some kind of message why compiling failed? I suppose that it can be difficult to recognize the reason for test failure without a message. Especially when you are new to a project or external contributor
> > 
> > 
> > I had it on my mind, but I decided to leave it with default formatting error that `scarb` displays (same when compilation fails due to any error, we don't handle it in script). Wdyt @kkawula ? Don't have a strong opinion though.
> 
> Hmm, I suppose that it will be helpful to at least add info about start of format checking, because at first look I don't know why this compilation failed [gh job](https://github.com/software-mansion/starknet.py/actions/runs/10579888230/job/29313414549). Some kind of `Checking contracts formatting` message

Sure, message that format checking has started will be helpful, will add it 👍

**Review by kkawula — APPROVED** [review]
LGMT

**Pull request merged**